src/app/rpmostree-override-builtins.cxx

```diff
@@ -89,8 +94,49 @@ handle_override (RPMOSTreeSysroot  *sysroot_proxy,
   g_autoptr(GVariant) options = g_variant_ref_sink (g_variant_dict_end (&dict));
 
   g_autoptr(GVariant) previous_deployment = rpmostree_os_dup_default_deployment (os_proxy);
+  g_autoptr(GPtrArray) override_replace_final = g_ptr_array_new();
+
+  if (override_replace && opt_ex_pin_from_repos)
+    {
+      g_autoptr(GPtrArray) queries = g_ptr_array_new();
+
+      for (const char *const* it = override_replace; it && *it; it++)
+        {
+          auto pkg = *it;
+
+          if(rpmostreecxx::is_http_arg(pkg) || rpmostreecxx::is_rpm_arg(pkg))
+            {
+              g_ptr_array_add(override_replace_final, (gpointer)pkg);
+              continue;
+	    }
+          else
+            {
+              g_ptr_array_add(queries, (gpointer)pkg);
+            }
+        }
+      if (queries->len > 0)
+        {
+          g_ptr_array_add(queries, NULL);
+          gchar **urls = NULL;
+
+          if (!rpmostree_os_call_refresh_md_sync(os_proxy, options, &transaction_address,
+                                                 cancellable, error))
+            return FALSE;
+          if (!rpmostree_transaction_get_response_sync(sysroot_proxy, transaction_address,
+                                                       cancellable, error))
+            return FALSE;
+          if (!rpmostree_osexperimental_call_search_sync (osexperimental_proxy,
+                                                          (const char *const*) queries->pdata,
+                                                          &urls, cancellable, error))
+            return FALSE;
+          for (const char *const* url = urls; url && *url; ++url)
+            {
+              g_ptr_array_add(override_replace_final, (gpointer)*url);
```

Comment:
I think it's OK if we don't want to `g_strdup`, though then we should move the `urls` declaration to the same level as `override_replace_final` so the values we get back from `Search()` live long enough.